snuba/web/rpc/v1/resolvers/R_eap_items/routing_strategies/storage_routing.py

```diff
@@ -0,0 +1,177 @@
+import uuid
+from dataclasses import dataclass, field
+from typing import Any, Callable, Dict, Optional, TypeAlias, Union
+
+import sentry_sdk
+from google.protobuf.json_format import MessageToDict
+from sentry_protos.snuba.v1.endpoint_time_series_pb2 import TimeSeriesRequest
+from sentry_protos.snuba.v1.endpoint_trace_item_table_pb2 import TraceItemTableRequest
+from sentry_protos.snuba.v1.request_common_pb2 import TraceItemType
+
+from snuba import environment
+from snuba.attribution import AppID
+from snuba.attribution.attribution_info import AttributionInfo
+from snuba.datasets.pluggable_dataset import PluggableDataset
+from snuba.downsampled_storage_tiers import Tier
+from snuba.query.logical import Query
+from snuba.query.query_settings import HTTPQuerySettings
+from snuba.request import Request as SnubaRequest
+from snuba.utils.metrics.timer import Timer
+from snuba.utils.metrics.util import with_span
+from snuba.utils.metrics.wrapper import MetricsWrapper
+from snuba.utils.registered_class import RegisteredClass
+from snuba.web import QueryResult
+from snuba.web.query import run_query
+
+_SAMPLING_IN_STORAGE_PREFIX = "sampling_in_storage_"
+_START_ESTIMATION_MARK = "start_sampling_in_storage_estimation"
+_END_ESTIMATION_MARK = "end_sampling_in_storage_estimation"
+MetricsBackendType: TypeAlias = Callable[
+    [str, Union[int, float], Optional[Dict[str, str]], Optional[str]], None
+]
+
+RoutedRequestType = Union[TimeSeriesRequest, TraceItemTableRequest]
+ClickhouseQuerySettings = Dict[str, Any]
+
+
+@dataclass
+class RoutingContext:
+    # NIT: some things should only be settable once
+    # e.g. query_result
+    in_msg: RoutedRequestType
+    timer: Timer
+    build_query: Callable[[TimeSeriesRequest | TraceItemTableRequest], Query]
```

Comment:
this is going to fail mypy, let's put `_build_query` as a method on the class and just import the build_query methods we need